### PR TITLE
Remove deprecated inner compaction type

### PR DESCRIPTION
**Pull request by jaime0815**
/kind improvement
Signed-off-by: yun.zhang <yun.zhang@zilliz.com>

### CODE REVIEW

File: internal/proto/data_coord.proto [hunk 1]
```diff
@@ -389,9 +389,8 @@ message ChannelWatchInfo {
 
 enum CompactionType {
   UndefinedCompaction = 0;
-  InnerCompaction = 1;
-  MergeCompaction = 2;
-  MixCompaction = 3;
+  MergeCompaction = 1;
+  MixCompaction = 2;
```
Comment (congqixia) [inline internal/proto/data_coord.proto:393]: `1` shall be left there to be compatible with older version.
We could just let enum value 1 to be reserved
[Reference](https://developers.google.com/protocol-buffers/docs/proto3#reserved_values)
``` protobuf
enum CompactType{
    reserved 2;
}
```

Comment (jaime0815) [reply internal/proto/data_coord.proto]: Thanks for your point out!

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**